changelog/3765.feature.rst

```diff
@@ -0,0 +1,5 @@
+Add support for composite entities.
+
+If you use ``from_entity`` in your custom slot mapping, you can now also specify a role and group label.
+If you set a role or group label, the slot is only filled if the entity has the specific role or group label set.
```

Comment:
And if you provide both, it has to match both right? (let me know if that use case doesn't even make sense)

Reply:
Yes, need to match both. This case is not very likely, but it might still happen.